src/Workspaces/Core/Portable/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -380,7 +380,7 @@ private static bool SymbolsAreCompatibleCore(ISymbol symbol, ISymbol newSymbol,
                     var type = methodSymbol.ContainingType;
                     var newType = newMethodSymbol.ContainingType;
                     if ((type != null && type.IsEnumType() &&
-                         type.EnumUnderlyingType != null && type.EnumUnderlyingType.SpecialType == newType.SpecialType) ||
+                         type.EnumUnderlyingType != null && type.EnumUnderlyingType.SpecialType == newType?.SpecialType) ||
```

Comment:
It looks like the condition below has the same issue, but with ```type```. Also, it looks like both ```type``` and ```newType``` must be not null in order for the whole condition to be true. Consider checking this fact for both types first and then continue with the ```||``` condition without any null checks for them.